pkg/health/store/store.go

```diff
@@ -0,0 +1,94 @@
+package store
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/square/p2/Godeps/_workspace/src/github.com/Sirupsen/logrus"
+
+	"github.com/square/p2/pkg/health"
+	"github.com/square/p2/pkg/health/checker"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/types"
+)
+
+// Health can answer questions about the health of a particular pod on a node
+// It performs this by watching the health tree of consul and caching the result
+type HealthStore interface {
+	StartWatch(quitCh <-chan struct{})
+	Fetch(types.PodID, string) *health.Result
+}
+
+type healthStore struct {
+	cachedHealth  map[cacheKey]*health.Result
+	healthChecker checker.ConsulHealthChecker
+	lock          *sync.RWMutex
+	logger        logging.Logger
+}
+
+var _ HealthStore = &healthStore{}
+
+func NewHealthStore(healthChecker checker.ConsulHealthChecker) HealthStore {
+	return &healthStore{
+		healthChecker: healthChecker,
+		lock:          &sync.RWMutex{},
```

Comment:
you should be able to make this not be a pointer and just use the zero value: "RWMutexes can be created as part of other structures; the zero value for a RWMutex is an unlocked mutex." (https://golang.org/pkg/sync/)